src/orquestra/sdk/_ray/_dag.py

```diff
@@ -238,16 +240,38 @@ class WfUserMetadata(t.TypedDict):
     workflow_def: t.Mapping[str, t.Any]
 
 
-def _get_python_packages(invocation: ir.TaskInvocation, wf: ir.WorkflowDef):
-    if not (deps_ids := wf.tasks[invocation.task_id].dependency_import_ids):
-        return []
+@singledispatch
+def _pip_string(_: ir.Import) -> t.List[str]:
+    return []
 
-    python_imports = [
-        imp
-        for imp_id in deps_ids
-        if isinstance((imp := wf.imports[imp_id]), ir.PythonImports)
+
+@_pip_string.register
+def _(imp: ir.PythonImports):
+    return [serde.stringify_package_spec(package) for package in imp.packages]
+
+
+@_pip_string.register
+def _(imp: ir.GitImport):
+    m = re.match(
+        r"(?P<user>.+)@(?P<domain>[^/]+?):(?P<repo>.+)", imp.repo_url, re.IGNORECASE
+    )
+    if m is not None:
+        url = f"ssh://{m.group('user')}@{m.group('domain')}/{m.group('repo')}"
+    else:
+        url = imp.repo_url
+    return [f"git+{url}@{imp.git_ref}"]
```

Comment:
**question**: I've seen you added an integration test for it. The coverage report haven't recorded testing this. Is it because this code is executed in another process?

Reply:
**suggestion**: could you also add a unit test for this function? Mistakes in regexps are easy to miss, and unit testing regexp-based code is usually easy to set up.

Reply:
@alexjuda yeah, this is executed inside Ray which doesn't get picked up by the coverage tool.
I'll add some unit tests for `_pip_string`